pkg/simulation/simulation.go

```diff
@@ -18,12 +20,13 @@ func NewSimulation(city *city.City) Simulation {
 func (s *Simulation) ResetTrams() {
 	s.trams = make([]*tram, len(s.city.GetTramTrips()))
 	for i, trip := range s.city.GetTramTrips() {
-		s.trams[i] = newTram(i, &trip)
+		s.trams[i] = newTram(i, &trip, s.controlCenter)
```

Comment:
Let's maybe pass the control center as a reference.
```suggestion
		s.trams[i] = newTram(i, &trip, &s.controlCenter)
```